pallets/identity/src/keys.rs

```diff
@@ -884,9 +884,34 @@ impl<T: Config> Module<T> {
         ensure_custom_length_ok::<T>(perms.complexity(), MAX_PERMISSION_COMPLEXITY)?;
         ensure_custom_length_ok::<T>(perms.asset.complexity(), MAX_ASSETS)?;
         ensure_custom_length_ok::<T>(perms.portfolio.complexity(), MAX_PORTFOLIOS)?;
+        Self::ensure_no_except_perms(&perms.extrinsic)?;
         Self::ensure_extrinsic_perms_length_limited(&perms.extrinsic)
     }
 
+    // Ensures that extrinsic permissions do not use the Except variant
+    // This is considered unsafe since extrinsic names can change or be replaced with newer versions
+    pub fn ensure_no_except_perms(perms: &ExtrinsicPermissions) -> DispatchResult {
+        let _ = match perms {
+            SubsetRestriction::Except(_) => {
+                return Err(Error::<T>::ExceptNotAllowedForExtrinsics.into());
+            }
+            SubsetRestriction::These(pallet_permissions) => {
+                for elem in pallet_permissions {
+                    match elem.dispatchable_names {
+                        SubsetRestriction::Except(_) => {
+                            return Err(Error::<T>::ExceptNotAllowedForExtrinsics.into());
+                        }
+                        _ => return Ok(()), // Matches Whole and These
```

Comment:
This would trigger the function to return `Ok` if the first element is not of type `SubsetRestriction::Except(_)` even if there is an Except restriction.

Reply:
Good catch.